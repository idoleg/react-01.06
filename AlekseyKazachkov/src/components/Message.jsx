import React from 'react';
import PropTypes from 'prop-types';
import "../styles/style.css";

export default class Message extends React.Component {
   static propTypes = {
       text: PropTypes.string.isRequired,
       sender: PropTypes.string.isRequired,
   };

    UNSAFE_componentWillMount() {
        // console.log('Message componentWillMount');
    }

    componentDidMount() {
        // console.log('Message componentDidMount');
    }

    componentDidUpdate() {
        // console.log('Message componentDidUpdate');
    }

   render() {
    // console.log("Message rander");
       return (
            <div className="message" style={{
                alignSelf: this.props.sender === "bot" ? "flex-start" : "flex-end"
            }}>
                <div className="message-sender">{this.props.sender}</div>
                <div>{ this.props.text }</div>
            </div>
        )
   }
}