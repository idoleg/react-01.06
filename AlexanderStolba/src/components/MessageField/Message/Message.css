.message {
    align-self: flex-end;
    list-style: none;
    background-color: #DC143C;
    padding: 6px;
    border-radius: 5px;
    width: 40%;
    font-size: 16px;
}

.message--is-robot {
    align-self: flex-start;
}

.message:not(:last-child) {
    margin-bottom: 10px;
}