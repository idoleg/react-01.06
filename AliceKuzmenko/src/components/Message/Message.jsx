import React from 'react';
import { ROBOT_NAME } from '../../containers/ChatContainer/ChatContainer';

export const Message = ({ name, content }) => {
    const isRobot = name === ROBOT_NAME ? ' robots-msg' : '';
    const isMe = name === 'Me' ? ' my-msg' : '';
    
return <li className={'message' + (isRobot) + (isMe)}><strong>{name}:</strong> {content}</li>
}