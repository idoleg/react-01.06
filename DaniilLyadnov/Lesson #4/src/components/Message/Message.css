.message {
	max-width: 400px;
	box-sizing: border-box;
	align-self: flex-start;
	background: #182533;
	font: 400 13px/18px 'Roboto';
	color: #fff;
	padding: 15px;
	border-radius: 5px;
	margin: 3px 0;
	word-break: break-word;
	position: relative;
}

.message:after {
	content: '';
	position: absolute;
	left: 0;
	bottom: 0;
	width: 0;
	height: 0;
	border: 15px solid transparent;
	border-right-color: #182533;
	border-left: 0;
	border-bottom: 0;
	margin-top: 0;
	margin-left: -11px;
}

.message strong {
	text-transform: uppercase;
}

.message:first-child {
	margin: 0 0 3px 0;
}

.message-is--robot {
	align-self: flex-end;
	background: #f1f1f1;
	color: #182533;
}

.message.message-is--robot:after {
	content: '';
	position: absolute;
	right: 0;
	left: unset;
	bottom: 0;
	width: 0;
	height: 0;
	border: 15px solid transparent;
	border-left-color: #f1f1f1;
	border-right: 0;
	border-bottom: 0;
	margin-top: 0;
	margin-right: -11px;
}