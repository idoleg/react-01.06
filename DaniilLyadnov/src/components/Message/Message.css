.message {
	align-self: flex-start;
	background: #e4ecf2;
	font: 13px/18px Tahoma, sans-serif, Arial, Helvetica;
	color: #000;
	padding: 7px 10px;
	border-radius: 3px;
	margin: 8px 0;
}

.message-is--robot {
	align-self: flex-end;
	background: #f1f1f1;
}