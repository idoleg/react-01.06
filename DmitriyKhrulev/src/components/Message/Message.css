.message {
   align-self: flex-end;
   list-style: none;
   background-color: lightblue;
   padding: 10px;
   border-radius: 5px;
   width: 30%;
}

.message:not(:last-child) {
   margin-bottom: 10px;
}

.message--is-robot {
   align-self: flex-start; 
}