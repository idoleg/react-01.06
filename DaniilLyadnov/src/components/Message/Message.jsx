import './Message.css';
import React from 'react';
import PropTypes from 'prop-types';
import {ROBOT_ROLE} from '../../containers/ChatContainer'

export const Message = ({name, content}) => {
    const isRobot = name === ROBOT_ROLE ? ' message-is--robot' : '';

    return (
        <li className={'message' + isRobot}><strong>{name}:</strong> {content}</li>
    )
};

Message.propTypes = {
    name: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
};