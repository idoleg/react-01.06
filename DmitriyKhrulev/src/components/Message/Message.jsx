import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import './Message.css'
import { ROBOT } from '../../store/botMiddleware'

export const Message = ({ name, content }) => {
   
   const isRobot = name === ROBOT ? 'message--is-robot' : '';
   return (<li className={'message ' + isRobot}><strong>{name}:</strong> {content}</li>);
};
Message.propTypes = {
   name: PropTypes.string.isRequired,
   content: PropTypes.string.isRequired,
}