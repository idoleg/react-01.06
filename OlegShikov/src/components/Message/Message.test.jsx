import React from 'react';
import { Message } from './Message';
import { shallow } from 'enzyme';

describe('Message.jsx', () => {
    it('render contetn', () => {
        const props = {
            name: "Ivan",
            content: "Hello!"
        }

        const element = shallow(<Message {...props} />);
        expect(element).toMatchSnapshot();
    })

    it('render message with robot css class', () => {
        const props = {
            name: "Robot",
            content: "Hello!"
        }

        const element = shallow(<Message {...props} />);
        expect(element).toMatchSnapshot();
    })
})