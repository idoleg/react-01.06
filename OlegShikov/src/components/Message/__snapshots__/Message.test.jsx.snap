// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message.jsx render contetn 1`] = `
<li
  className="message"
>
  <strong>
    Ivan
    :
  </strong>
   
  Hello!
</li>
`;

exports[`Message.jsx render message with robot css class 1`] = `
<li
  className="message message--is-robot"
>
  <strong>
    Robot
    :
  </strong>
   
  Hello!
</li>
`;
